### PR TITLE
2.0 selection demos

### PR DESCRIPTION
**Pull request by manolo**
Connected to #519

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/569)

### CODE REVIEW

**Comment by Saulis** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[docs/vaadin-grid-selection.adoc, line 11 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYrWWTpo7CWi-XGEgLh:-KYrWWTpo7CWi-XGEgLi:b-2i8y98) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7efad2c97faf6de8e7f435adc1dd9bc6bfe8a4e6/docs/vaadin-grid-selection.adoc#L11)):*
> ```AsciiDoc
> 
> Grid allows item based selection of data rows either through UI interaction or using the JavaScript API.
> The selection behavior depends on the currently active selection mode.
> ```

As we're dropping the concept of "selection mode", this line can be removed.

---

*[docs/vaadin-grid-selection.adoc, line 14 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYrWs03cZH1K2j6psxb:-KYrWs03cZH1K2j6psxc:b65zgm1) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7efad2c97faf6de8e7f435adc1dd9bc6bfe8a4e6/docs/vaadin-grid-selection.adoc#L14)):*
> ```AsciiDoc
> 
> [[vaadin-grid.selection.mode]]
> == Selection Mode
> ```

I think we should have four main topics here: a) displaying and modifying the selection state using `[[selected]]` and `{{selected}}` in template bindings, b) using `<vaadin-grid-selection-column>` to get default templates and functionality for the selection, c) customising the selection behavior d) using the selection API

I'm not sure about what's the best order for a) and b) – we can either direct people to use `<vaadin-grid-selection-column>` as the default which they can customise OR we can say the use the basic `[[selected]]` bindings by default, but for most basic cases we have this helper element for you?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569)*

**Comment by manolo** [comment]
Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[docs/vaadin-grid-selection.adoc, line 11 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYrWWTpo7CWi-XGEgLh:-KYvy9ofIOVqv4MPQlco:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7efad2c97faf6de8e7f435adc1dd9bc6bfe8a4e6/docs/vaadin-grid-selection.adoc#L11)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

As we're dropping the concept of "selection mode", this line can be removed.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 14 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYrWs03cZH1K2j6psxb:-KYsW8nFnZBpHPQuPn4H:b-12nvrt) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7efad2c97faf6de8e7f435adc1dd9bc6bfe8a4e6/docs/vaadin-grid-selection.adoc#L14)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

I think we should have four main topics here: a) displaying and modifying the selection state using `[[selected]]` and `{{selected}}` in template bindings, b) using `<vaadin-grid-selection-column>` to get default templates and functionality for the selection, c) customising the selection behavior d) using the selection API

I'm not sure about what's the best order for a) and b) – we can either direct people to use `<vaadin-grid-selection-column>` as the default which they can customise OR we can say the use the basic `[[selected]]` bindings by default, but for most basic cases we have this helper element for you?
</blockquote></details>

a) b) d) done.

But I'm not sure about c) we already provide an demo for custom selection behavior which seems so long to add again in the documentation. I have added a simpler one for this case. Using events would be simpler, but the `selected-items-changed` looks quite complex for the user.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569)*

**Comment by Saulis** [comment]
Review status: 0 of 4 files reviewed at latest revision, 17 unresolved discussions.

---

*[docs/vaadin-grid-selection.adoc, line 8 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx68P-60H_TVp92Usl:-KYx68P-60H_TVp92Usm:b4o07z8) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L8)):*
> ```AsciiDoc
> 
> [[vaadin-grid.selection.binding]]
> == Binding to the Selection State.
> ```

Titles should not end with a period since they're not sentences.

---

*[docs/vaadin-grid-selection.adoc, line 10 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx6Kvm025cRiy5kKhN:-KYx6Kvm025cRiy5kKhO:b-2bi0cp) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L10)):*
> ```AsciiDoc
> == Binding to the Selection State.
> 
> The [vaadinelement]#vaadin-grid# implements multi selection behavior by default.
> ```

"The" isn't needed since we're not referring to a specific instance of vaadin-grid.

---

*[docs/vaadin-grid-selection.adoc, line 11 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx6bIQe64ZQGv7dA9E:-KYx6bIQe64ZQGv7dA9F:b649982) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L11)):*
> ```AsciiDoc
> 
> The [vaadinelement]#vaadin-grid# implements multi selection behavior by default.
> But you need a mechanisim to allow the user to select rows. It can be done by adding a column that provides an element to bind the `selected` property of the row.
> ```

"Two-way binding `selected` variable inside templates can be used to allow users to select rows."

---

*[docs/vaadin-grid-selection.adoc, line 15 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx6vIIgyggKRVVBSkV:-KYx6vIIgyggKRVVBSkW:b-3tkaez) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L15)):*
> ```AsciiDoc
> [source,html]
> ----
> <link rel="import" href="../vaadin-grid-selection-column.html">
> ```

vaadin-grid-selection-column isn't used in this example. In general, we haven't added any import lines in our documentation examples. I can see two reasons: 1) they need to be complete, meaning importing also vaadin-grid.html and other element, which in turn will bloat the example draw space and attention from the actual context. 2) the import path depend on users context, so it would be misleading to say it's always in the form of "../vaadin-grid/vaadin-grid.html" etc.

IMO it's just overall better to leave the imports out. It's good to have the note section, which you added, to make user aware of the fact that `vaadin-grid-selection-column.html` needs to be imported separately in case they don't expect it.

---

*[docs/vaadin-grid-selection.adoc, line 19 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx8ObY4o_GDI-QOZXY:-KYx8ObY4o_GDI-QOZXZ:boiea2c) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L19)):*
> ```AsciiDoc
> <vaadin-grid items='["One", "Two", "Tree"]'>
>   <vaadin-grid-column>
>     <input type="checkbox" checked="{{selected::change}}" />
> ```

this column is missing `<template>` elements. Also, I would prefer using `<paper-checkbox checked="{{selected}}">` as a simpler example of two-way binding vs. native elements.

---

*[docs/vaadin-grid-selection.adoc, line 22 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx8m4L3tfPPQWHJwRl:-KYx8m4L3tfPPQWHJwRm:b-h0583u) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L22)):*
> ```AsciiDoc
>   </vaadin-grid-column>
>   <vaadin-grid-column width="50px">
>     <template class="header">Number</template>
> ```

This column is missing a cell template

---

*[docs/vaadin-grid-selection.adoc, line 28 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9ByDnCv8YnqwXwo5:-KYx9ByDnCv8YnqwXwo6:bfidd8y) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L28)):*
> ```AsciiDoc
> 
> [[vaadin-grid.selection.column]]
> == The Selection Column Helper element.
> ```

Would rephrase this to "Using the Selection Column"

---

*[docs/vaadin-grid-selection.adoc, line 32 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9Jdsc6ywuLs4fV_O:-KYx9Jdsc6ywuLs4fV_P:b-zho2w6) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L32)):*
> ```AsciiDoc
> You can use the [vaadinelement]#vaadin-grid-selection-column# helper element to get default templates and functionality for the selection. With it the user can select rows using checkboxes displayed in a column.
> 
> In the case that the grid is configured with an array of items as the data source, the [vaadinelement]#vaadin-grid-selection-column# provides the feature of selecting all the items by clicking on the checkbox at the header.
> ```

"When [vaadinelement]#vaadin-grid# is configured..."

and

"...by clicking on the checkbox in the header:"

---

*[docs/vaadin-grid-selection.adoc, line 36 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9_LBoTLL2D73BGBW:-KYx9_LBoTLL2D73BGBX:biqsuwu) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L36)):*
> ```AsciiDoc
> [source,html]
> ----
> <link rel="import" href="../vaadin-grid-selection-column.html">
> ```

I would drop the import

---

*[docs/vaadin-grid-selection.adoc, line 53 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxBxuOSpP8gVOtd2ci:-KYxBxuOSpP8gVOtd2cj:b-9a68na) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L53)):*
> ```AsciiDoc
> image::img/vaadin-grid-selection-column.png[]
> 
> In addition, you can customize the selection column, changing it size, displaying it at any position in the grid, or providing your own template for checkboxes.
> ```

".. customize the selection column by changing its width, flex, frozen properties, or by providing your own templates." 

I don't consider placing the it in different place "customizing" the selection column itself.

---

*[docs/vaadin-grid-selection.adoc, line 58 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9h8lsbkR5HQXTB0y:-KYx9h8lsbkR5HQXTB0z:b-ebhteq) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L58)):*
> ```AsciiDoc
> ----
> <link rel="import" href="../vaadin-grid/vaadin-grid-selection-column.html">
> <link rel="import" href="../polymer-checkbox/polymer-checkbox.html">
> ```

"paper-checkbox" mispelled, I would drop both imports

---

*[docs/vaadin-grid-selection.adoc, line 87 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxAQ41qm_v3DUqZHoI:-KYxAQ41qm_v3DUqZHoJ:b-q060r5) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L87)):*
> ```AsciiDoc
> 
> [methodname]#grid.selectItem(item)#::
>   Selects the row with the given item. It sets the property `selected` to true, adds the attribute to the row, and updates appropriately the `selectedItems` array.
> ```

I don't think we should go into implementation details. We should keep a healthy level of abstraction here. If we say that calling `selectItem()` will select an item and in a separate paragraph tell that `selectedItems` holds all the selected items, it's pretty obvious that selecting an item will update the `selectedItems`.

---

*[docs/vaadin-grid-selection.adoc, line 93 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxB6G0ZL56762xd2sa:-KYxB6G0ZL56762xd2sb:bqysa0m) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L93)):*
> ```AsciiDoc
> 
> [propertyname]#selectedItems#::
>   It's the property that represents the list of selected items. You can either use `selectItem` or `deselectItem` to modify the list, or modify the array directly by using Polymer API.
> ```

"Array of currently selected items. You can either..."

---

*[docs/vaadin-grid-selection.adoc, line 95 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxBGBCvm3AyXUe1n1o:-KYxBGBDf5Zl7f50dU7U:b-i519o2) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L95)):*
> ```AsciiDoc
>   It's the property that represents the list of selected items. You can either use `selectItem` or `deselectItem` to modify the list, or modify the array directly by using Polymer API.
> 
> [propertyname]#selected-items-changed::
> ```

As this isn't a custom event, but instead a normal Polymer change notification event, I don't see any reason to explicitly mention it here. We don't mention notification events for any other properties either.

---

*[docs/vaadin-grid-selection.adoc, line 108 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxBe2VpQxuSe243kPX:-KYxBe2VpQxuSe243kPY:b-9qvn3k) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L108)):*
> ```AsciiDoc
> 
> [[vaadin-grid.selection.model]]
> == Customize the Selection Model
> ```

"Customizing the Selection behavior"

---

*[docs/vaadin-grid-selection.adoc, line 111 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxDTLj7QmZNWx8v1B6:-KYxDTLj7QmZNWx8v1B7:bebgxt1) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L111)):*
> ```AsciiDoc
> 
> By default, the selection behavior of the grid uses a multiselection mode stored in the `selectedItems` array.
> If you wanted to modify the behavior you need to use the selection API to deselect the previo.
> ```

"In case you want to modify the default selection behavior, you can interact directly with `selectedItems` array."

(I don't think it's necessary here to repeat the "multi selection mode" since you've mentioned it already in the very first chapter as being the default.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569)*

**Comment by manolo** [comment]
Review status: 0 of 4 files reviewed at latest revision, 17 unresolved discussions.

---

*[docs/vaadin-grid-selection.adoc, line 8 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx68P-60H_TVp92Usl:-KZ0LcFKtTZIlvHaw6_6:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L8)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Titles should not end with a period since they're not sentences.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 10 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx6Kvm025cRiy5kKhN:-KZ0LhAvDd4q3vqru2e5:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L10)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

"The" isn't needed since we're not referring to a specific instance of vaadin-grid.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 11 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx6bIQe64ZQGv7dA9E:-KZ0MCp9Vk4niFl170UF:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L11)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

"Two-way binding `selected` variable inside templates can be used to allow users to select rows."
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 15 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx6vIIgyggKRVVBSkV:-KZ0MQMnhGXu9GRPTZny:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L15)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

vaadin-grid-selection-column isn't used in this example. In general, we haven't added any import lines in our documentation examples. I can see two reasons: 1) they need to be complete, meaning importing also vaadin-grid.html and other element, which in turn will bloat the example draw space and attention from the actual context. 2) the import path depend on users context, so it would be misleading to say it's always in the form of "../vaadin-grid/vaadin-grid.html" etc.

IMO it's just overall better to leave the imports out. It's good to have the note section, which you added, to make user aware of the fact that `vaadin-grid-selection-column.html` needs to be imported separately in case they don't expect it.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 19 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx8ObY4o_GDI-QOZXY:-KZ0Mhy4v252SsiQoSYm:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L19)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

this column is missing `<template>` elements. Also, I would prefer using `<paper-checkbox checked="{{selected}}">` as a simpler example of two-way binding vs. native elements.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 22 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx8m4L3tfPPQWHJwRl:-KZ0N2qtMewChf8g_YBR:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L22)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

This column is missing a cell template
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 28 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9ByDnCv8YnqwXwo5:-KZ0ObdvBjXPztSWzl0x:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L28)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Would rephrase this to "Using the Selection Column"
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 32 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9Jdsc6ywuLs4fV_O:-KZ0NKZEWD722ETT7jyU:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L32)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

"When [vaadinelement]#vaadin-grid# is configured..."

and

"...by clicking on the checkbox in the header:"
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 36 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9_LBoTLL2D73BGBW:-KZ0NNbgUvc3Uowm35dK:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L36)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

I would drop the import
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 53 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxBxuOSpP8gVOtd2ci:-KZ0NXvCxNW4V8RqE_lK:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L53)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

".. customize the selection column by changing its width, flex, frozen properties, or by providing your own templates." 

I don't consider placing the it in different place "customizing" the selection column itself.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 58 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYx9h8lsbkR5HQXTB0y:-KZ0NcMjwG9Kq57HPhLY:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L58)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

"paper-checkbox" mispelled, I would drop both imports
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 87 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxAQ41qm_v3DUqZHoI:-KZ0Nzs2SYVLVYQmiO52:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L87)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

I don't think we should go into implementation details. We should keep a healthy level of abstraction here. If we say that calling `selectItem()` will select an item and in a separate paragraph tell that `selectedItems` holds all the selected items, it's pretty obvious that selecting an item will update the `selectedItems`.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 93 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxB6G0ZL56762xd2sa:-KZ0OB9B7V76OK8DzV45:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L93)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

"Array of currently selected items. You can either..."
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 95 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxBGBCvm3AyXUe1n1o:-KZ0OKlwXTZ536GPDH9y:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L95)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

As this isn't a custom event, but instead a normal Polymer change notification event, I don't see any reason to explicitly mention it here. We don't mention notification events for any other properties either.
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 108 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxBe2VpQxuSe243kPX:-KZ0OOkIbIdAaYC2tDh7:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L108)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

"Customizing the Selection behavior"
</blockquote></details>

Done.

---

*[docs/vaadin-grid-selection.adoc, line 111 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-KYxDTLj7QmZNWx8v1B6:-KZ0OY9B95J3mHAcrL8J:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/7f7ad87eaf502eafb84f30c27953432d2b63ea34/docs/vaadin-grid-selection.adoc#L111)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

"In case you want to modify the default selection behavior, you can interact directly with `selectedItems` array."

(I don't think it's necessary here to repeat the "multi selection mode" since you've mentioned it already in the very first chapter as being the default.)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569)*

**Comment by Saulis** [comment]
:lgtm:

---

Reviewed 3 of 4 files at r1, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/569#-:-KZ0VaIhSxSFPDwGxM9n:bnfp4nl)*
